src/utils/dom-fingerprint.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+
+/**
+ * Gets a string showing the index of an element within
+ * the children of its parent, counting only nodes with the same tag.
+ * Stop at 25, just to have a limit.
+ * @param {?Element} element DOM node to get index of.
+ * @return {string} '.<index>' or ''.
+ */
+function indexWithinParent(element) {
+  if (element && element.nodeName && element.parentElement) {
+    const elementParent = element.parentElement;
+    const nodeName = element.nodeName.toString().toLowerCase();
```

Comment:
Sadly, some browser returns an object for nodeName. I don't recall which. But it's probably an older one; certainly not a mobile one. I will add a comment here.